game/plugin/skills/runecrafting/src/action.kt

```diff
@@ -0,0 +1,70 @@
+import org.apollo.cache.def.ItemDefinition
+import org.apollo.game.action.ActionBlock
+import org.apollo.game.action.AsyncDistancedAction
+import org.apollo.game.action.DistancedAction
+import org.apollo.game.model.Position
+import org.apollo.game.model.entity.Player
+import org.apollo.game.plugin.api.runecraft
+
+class TeleportAction(val player: Player, val start: Position, val distance: Int, val end: Position): DistancedAction<Player>(0, true, player, start, distance) {
+    override fun executeAction() {
+        player.teleport(end)
+        stop()
+    }
+}
+
+class CreateTiaraAction(val player: Player, val position: Position, val tiara: Tiara, val alter: Alter): DistancedAction<Player>(0, true, player, position, 2) {
+    override fun executeAction() {
+        //Check if correct alter
+        if (tiara.alter != alter) {
+            player.sendMessage("You can't use that talisman on this alter.")
+            stop()
```

Comment:
Shouldn't there be a return here to prevent the player from still getting to the next if statement?

Reply:
Yes! I figured out how to do it. Other plugins might also have this problem, including but not limited to fishing.

Reply:
I am actually not sure. Looking at other plugins, I do not see any return inside of an `ActionBlock`, I am only seeing `stop()`

Reply:
`stop()` inside the scope of an `ActionBlock` (or any other kind of asynchronous action) is suspendable, meaning it'll immediately suspend execution from that block.  You don't need a return there.